File: nim_core_v2/ios/Classes/ResultCallback.swift
// Copyright (c) 2022 NetEase, Inc. All rights reserved.
// Use of this source code is governed by a MIT license that can be
// found in the LICENSE file.

import Foundation

class ResultCallback {
  private var safeResult: SafeResult?
  init(_ result: @escaping FlutterResult) {
    safeResult = SafeResult(result)
  }

  func result(_ result: Any) {
    safeResult?.result(result)
  }

  func notImplemented() {
    safeResult?.notImplemented()
  }
}

File: nim_core_v2/android/src/main/kotlin/com/netease/nimflutter/ResultCallback.kt
/*
 * Copyright (c) 2022 NetEase, Inc. All rights reserved.
 * Use of this source code is governed by a MIT license that can be
 * found in the LICENSE file.
 */

package com.netease.nimflutter

class ResultCallback<T>(private val safeResult: SafeResult) {
    fun result(data: NimResult<T>) {
        safeResult.success(data.toMap())
    }
}
